resolveConfig.d.ts

```diff
@@ -1,3 +1,12 @@
-import type { Config } from './types/config'
-declare function resolveConfig(config: Config): Config
+import type { Config, ResolvableTo } from './types/config'
+
+type UnwrapResolvables<T> = {
+  [K in keyof T]: T[K] extends ResolvableTo<infer R> ? R : T[K]
+}
+
+type ResolvedConfig<T extends Config> = Omit<T, 'theme'> & {
+  theme: UnwrapResolvables<T['theme']>
```

Comment:
You're gonna want to apply the same unwrapping to all the keys inside `theme.extend`

Reply:
Actually, let me re-word this. You need to unwrap all the keys inside theme.extend and then spread them into the top level object. Is that even possible??

Reply:
Scratch this. It'll require some ridiculousness with conditional types and what not. Nevermind.